src/rgw/rgw_rados.cc

```diff
@@ -12052,3 +12052,45 @@ librados::Rados* RGWRados::get_rados_handle()
   }
 }
 
+int RGWRados::delete_obj_aio(rgw_obj& obj, rgw_bucket& bucket,
+                              RGWBucketInfo& bucket_info, RGWObjState *astate,
+                              list<librados::AioCompletion *>& handles)
+{
+  rgw_rados_ref ref;
+  int ret = get_obj_ref(obj, &ref, &bucket);
+  if (ret < 0) {
+    lderr(cct) << "ERROR: failed to get obj ref with ret=" << ret << dendl;
+    return ret;
+  }
+
+  RGWRados::Bucket bop(this, bucket_info);
+  RGWRados::Bucket::UpdateIndex index_op(&bop, obj, astate);
+
+  ObjectWriteOperation op;
+  ret = index_op.prepare(CLS_RGW_OP_DEL);
```

Comment:
@prallabh the idea is to avoid accessing the bucket index altogether at this point. If you're removing the entire bucket, there's no reason to update the bucket index for every object. Just remove the objects and delete the bucket index at the end.